packages/contracts-bedrock/test/properties/helpers/MockL2ToL2CrossDomainMessenger.t.sol

```diff
@@ -37,22 +55,64 @@ contract MockL2ToL2CrossDomainMessenger {
         superTokenInitDeploySalts[token] = deploySalt;
     }
 
+    function messageQueueLength() public view returns (uint256) {
+        return _messageQueue.length;
+    }
+
+    function setAtomic(bool atomic) public {
+        _atomic = atomic;
+    }
+
+    function relayMessageFromQueue(uint256 rawIndex) public {
+        uint256 index = rawIndex % _messageQueue.length;
+        CrossChainMessage memory message = _messageQueue[index];
+        _messageQueue[index] = _messageQueue[_messageQueue.length - 1];
```

Comment:
Missing check `_length > 0` here so it doesn't revert due an underflow when subtracting 1?

Reply:
Actually, I think that check should go on the first line, since if `0`, the modulo operation will revert

Reply:
medusa (and foundry, most of the time) ignore reverts in invariant tests, and only consider `assert(false)`, aka `Panic(0x01)`, as test failures. That's why we `try` and `catch` on function calls where we want to fail the test if it reverts: because otherwise it is ignored.

coverage reports are then used to know if tests are reverting too early and leaving assertions or state transitions uncovered. There are many other places throughout the campaign where execution might return early due to a revert, and many places where we explicitly revert instead of returning because it's easier :D

Reply:
Oh amazing, good to know :)